src/docket/worker.py

```diff
@@ -377,6 +377,9 @@ async def _execute(self, message: RedisMessage) -> None:
 
         execution = Execution.from_message(function, message)
 
+        async with self.docket.redis() as redis:
+            await redis.delete(self.docket.known_task_key(execution.key))
```

Comment:
I spent a lot of time mulling this placement.  It's pretty close to the start of execution, but I think that it's safe as long as it happens any time before the task is `XACK`ed.  If anything goes wrong between retrieving the task from the stream and deleting this key here, the task will be redelivered and we'll eventually delete this key on a later attempt (as long as workers are running).

I'm mostly worried about a key getting "stuck" in a way that prevents any future scheduling of it without manual intervention, rather than accidentally allowing a task to get rescheduled with the same key.